fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java

```diff
@@ -315,8 +335,12 @@ public static void validateDefaultValue(Type type, String defaultValue) throws A
                 decimalLiteral.checkPrecisionAndScale(scalarType.getScalarPrecision(), scalarType.getScalarScale());
                 break;
             case DATE:
+                DateLiteral dateLiteral = new DateLiteral(defaultValue, type);
+                break;
             case DATETIME:
-                new DateLiteral(defaultValue, type);
+                if (defaultValueExprDef == null || !defaultValueExprDef.getExprName().equals(DefaultValue.NOW)) {
+                    DateLiteral datetimeLiteral = new DateLiteral(defaultValue, type);
```

Comment:
if defaultValueExprDef == null, then check datetimeLiteral
if defaultValueExprDef != null, then check the function name == now.